src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,181 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * TaskList class that contains an ArrayList of tasks
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructor for TaskList class, initialises a new ArrayList
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<Task>();
+    }
+
+    /**
+     * prints out the list
+     */
+    public void printList() {
+        for (int i = 1; i <= tasks.size(); i++) {
+            Task task = tasks.get(i - 1);
+            System.out.print(i + ": ");
+            task.printTask();
+        }
+    }
+
+    public ArrayList<Task> getTasks() {
+        return this.tasks;
+    }
+
+    public int getSize() {
+        return tasks.size();
+    }
+
+    /**
+     * Clears the list of tasks
+     */
+    public void reset() {
+        tasks.clear();
+    }
+
+    /**
+     * Takes in a String input of a task to be added from the user that is passed by parser
+     *
+     * @param input String that contains task type, task details and task date/time if applicable
+     */
+    //Add from String input by user
+    public void addTask(String input) {
```

Comment:
Any reason why you included this method as the parsing of the input to create a task is already being handled by the Parser class?